packages/plugins/graphql/server/services/internals/types/response-collection-meta.js

```diff
@@ -25,9 +25,9 @@ module.exports = ({ strapi }) => {
             const { start, limit } = args;
 
             const total = await strapi.entityService.count(resourceUID, args);
-            const pageSize = limit;
-            const pageCount = limit === 0 ? 0 : Math.ceil(total / limit);
-            const page = limit === 0 ? 1 : Math.floor(start / limit) + 1;
+            const pageSize = limit === -1 ? total : limit;
```

Comment:
I was wondering if it wouldn't be better to set the pageSize to `(limit || total) - start` (total if limit === -1 else limit). I think it would better reflect the actual size of the returned page.
What do you think?